overlord.pl

```diff
@@ -391,8 +391,8 @@ sub squidHasAllKids
     # TODO: If there is a disker, we should (also) wait for kids to find the
     # disker strand (mtFindStrand). Is there a mgr page reflecting that state?
     my $mgrPage = &getCacheManagerResponse('openfd_objects')->{content};
-    # how many kids completed their by kidN {...} by kidN reports
-    my $kidsRegistered = () = $mgrPage =~ /^[}] by kid\d+/mg;
+    # how many kids completed their per-kid reports
+    my $kidsRegistered = () = $mgrPage =~ /^([}] by kid\d+|[.]{3})/mg;
```

Comment:
As of now, this PR is not sufficient to accommodate current https://github.com/squid-cache/squid/pull/1784 code (as of that PR commit [95f24ece](https://github.com/squid-cache/squid/pull/1784/commits/95f24ece51217ea65d4c80cce53694c53f71a9aa)) because the latter PR also affects kidsRegistered vs. kidsExpected math by adding `by kid` framing to non-SMP Squids, increasing kidsRegistered values for those Squids. The changes in non-SMP kidsRegistered values trigger a die() assertion a few line below.

I believe I know what code changes are needed, but they will need to be done in both Daft and Squid Overlord repositories. I am not yet sure whether we can easily and safely support both current and post-1784 Squids with the same Daft/Overlord code. I hope to avoid this, but it may be necessary to require other Squid PRs to incorporate PR 1784 changes after that PR merges into Squid master/v7.

Reply:
> The changes in non-SMP kidsRegistered values trigger a die() assertion a few lines below.

Fixed in branch commit 8a835fbc.